### PR TITLE
GLSP-1116 Revise model loading

### PR DESCRIPTION
**Pull request by tortmayr**
# GLSP-1116 Revise model loading
- Refactor `ModelSubmissionHandler` to support proper handling of the `RequestModelAction` as real request action 
- Add support for rejecting request actions
 Part-of: https://github.com/eclipse-glsp/glsp/issues/1116 
 Part-of: https://github.com/eclipse-glsp/glsp/issues/606
 
# GLSP-1117: Remove need for explicit definition of client actions
 Refactor the base GLSP protocol to allow the client to tell the server which actions it is going to handle i.e. which actions should be forwarded to the client
- Add `clientActions` array to `InitializeClientSessionParams`. This means the client now has to pass the action kinds it wants to handle as part of the initalize request
- Refactor `ClientSessionManager` API directly use the `InitializeClientSessionParams` object for creating new sessions. This means that the `ClientSessionManager` can also access the generic `args` properties that might have been passed with the initialize request.
- Replace `ClientActionHandler` with `ClientActionForwader` a separate component that is not part of the server-side action handlers. 
- Remove `configureClientActions` method from `DiagramModule` as the explicit configuration is no longer needed
- Refactor `ClientIdModule` to `ClietnSessionModule` responsible for injection session specific configuration like the clientId and the clientActions

Part of eclipse-glsp/glsp/issues/1117

# GLSP-1071: Rename ServerStatus/ServerMessage action
Also remove unused `ServerStatus` class
Part of https://github.com/eclipse-glsp/glsp/issues/1071

### CODE REVIEW

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/actions/RejectAction.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.actions;
+
+import java.util.Optional;
+
+/**
+ * A reject action is a {@link ResponseAction} fired to indicate that a certain {@link ResponseAction}
```
Comment (planger) [inline plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/actions/RejectAction.java:21]:
```suggestion
 * A reject action is a {@link ResponseAction} fired to indicate that a certain {@link RequestAction}
```

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java [hunk 2]
```diff
@@ -37,6 +37,15 @@
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
+/**
+ * Helper class that provide utility methods to handle model updates i.e.
```
Comment (planger) [inline plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java:41]:
```suggestion
 * Helper class that provides utility methods to handle model updates i.e.
```

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java [hunk 3]
```diff
@@ -37,6 +37,15 @@
...
+ * submit a new model to the client. In addition, to the core model update action this clas
```
Comment (planger) [inline plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java:42]:
```suggestion
 * submit a new model to the client. In addition, to the core model update action this class
```

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java [hunk 4]
```diff
@@ -37,6 +37,15 @@
...
+ * also takes care related behavior like dirty state handling, validation and client/server side layouting.
```
Comment (planger) [inline plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java:43]:
```suggestion
 * also takes care of related behavior like dirty state handling, validation and client/server side layouting.
```

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java [hunk 5]
```diff
@@ -37,6 +37,15 @@
...
+ * Note that the submissions handler is only responsible for deriving the set of actions that comprise a model update
+ * but does not actually dispatch them. The returned actions but should be either manually dispatched
```
Comment (planger) [inline plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java:45]:
```suggestion
 * but does not actually dispatch them. The returned actions should be either manually dispatched
```

**Review by planger — APPROVED** [review]
Thanks, looks good to me! :+1: Just a few minor doc suggestions inline.

**Comment by tortmayr** [comment]
@planger  I have also added change for GLSP-1117 and GLSP-1071 to this PR. Since these are also changes that require updatse across all integration projects I would rather do this all in one go instead of having to create multiple PRs per repository.

Could you please re-review?

**Review by planger — APPROVED** [review]
Thanks, looks good to me! :+1:

**Pull request merged**